### PR TITLE
[GH-253] Add filepath selection based on OS for nftables.conf

### PR DESCRIPTION
**Pull request by danielpkrueger**
# Description

Updates nftables recipe to support Oracle Linux 8 file paths.

## Issues Resolved

[GH-253]

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: resources/nftables.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@
 action :rebuild do
   ensure_default_rules_exist(new_resource)
 
-  file '/etc/nftables.conf' do
+  file nftables_conf_location do
```
Comment (Stromweld) [inline resources/nftables.rb:43]: @bmhughes was asking for this to be based on the property and the default property value would then be based on this helper method. That way if the OS isn't supported yet users can manually specify the path through setting the property.

so this line would look something like `file new_resource.nftables_conf_path do` and then around line 32 you'd have the nftables_conf_path property defined.

**Review by Stromweld — CHANGES_REQUESTED** [review]

File: resources/nftables.rb (hunk 1 above)
Comment (bmhughes) [reply resources/nftables.rb:43]: To add to this, an example of how the property definition should look [here](https://github.com/sous-chefs/nginx/blob/d1b9d7be4a4b4bc41c8283ce928b40073fd05a22/resources/config.rb#L71)

File: resources/nftables.rb [hunk 2]
```diff
@@ -1,9 +1,7 @@
 unified_mode true
 
-action_class do
-  include FirewallCookbook::Helpers
-  include FirewallCookbook::Helpers::Nftables
-end
+include FirewallCookbook::Helpers
+include FirewallCookbook::Helpers::Nftables
```
Comment (danielpkrueger) [inline resources/nftables.rb:4]: @Stromweld @bmhughes I was tinkering with creating property, but ran into undefined methods messages until I removed the includes from the action_class to their own lines. I'm not sure if my change is the proper way to go about this and would love any insight you might have. I've been reviewing chef.io docs, but haven't found a specific answer yet.

Comment (Stromweld) [reply resources/nftables.rb:4]: That makes sense as action class would only include it for the different action sections. Moving it out should make it available to the properties and actions I believe.

**Comment by Stromweld** [comment]
@Xorima, @ramereth would either of you know why the validators aren't running?

File: resources/nftables.rb (hunk 2 above)
Comment (bmhughes) [reply resources/nftables.rb:4]: Looks like they were `include`'d before but not actually used in the action class, if they were then you'd need both but for the time we an leave it like this.

**Comment by Xorima** [comment]
Fixed @Stromweld for now, as per slack I need to put in a long term fix

**Review by Stromweld — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [6.1.0](https://github.com/sous-chefs/firewall/releases/tag/6.1.0)

**Pull request merged**